components/brave_rewards/common/publisher_utils_unittest.cc

```diff
@@ -0,0 +1,79 @@
+/* Copyright (c) 2024 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at https://mozilla.org/MPL/2.0/. */
+
+#include "brave/components/brave_rewards/common/publisher_utils.h"
+
+#include <optional>
+#include <string>
+
+#include "testing/gtest/include/gtest/gtest.h"
+#include "url/gurl.h"
+
+namespace brave_rewards {
+
+class RewardsPublisherUtilsTest : public testing::Test {
+ protected:
+  std::optional<std::string> GetPublisherId(const std::string& url) {
+    return GetPublisherIdFromURL(GURL(url));
+  }
+};
+
+TEST_F(RewardsPublisherUtilsTest, IsMediaPlatformURL) {
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("https://brave.com")));
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("http://brave.com")));
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("https://search.brave.com")));
+
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("https://brave.co.uk")));
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("https://www.brave.co.uk")));
+
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("file:///a/b/c/")));
+  EXPECT_FALSE(IsMediaPlatformURL(GURL("invalid-url")));
+
+  EXPECT_TRUE(IsMediaPlatformURL(GURL("https://twitter.com/foo")));
+  EXPECT_TRUE(IsMediaPlatformURL(GURL("https://www.twitter.com/foo")));
```

Comment:
Could we also add the same checks for x.com, just for completeness?

Reply:
Done.